polars/polars-core/src/series/mod.rs

```diff
@@ -627,6 +625,16 @@ impl Series {
         }
     }
 
+    /// Count the number of elements elements over an expanding window.
+    pub fn cumcount(&self, reverse: bool, count_nulls: bool) -> Series {
+        let flags = if count_nulls {
+            self.is_null() | !self.is_null()
+        } else {
+            !self.is_null()
+        };
+        flags.into_series().cumsum(reverse)
+    }
```

Comment:
I'm SURE this is not the most efficient way to do this... suggestions welcome

Reply:
I have a strategy for improving this, but I can't get it to compile

```rust
fn det_count<T>(count: &mut u32, x: Option<T>) -> Option<Option<u32>> {
    if let Some(_) = x {
        *count += 1
    }
    Some(Some(*count))
}


impl<T> ChunkCumAgg<T> for ChunkedArray<T>
where
    T: PolarsNumericType,
    ChunkedArray<T>: FromIterator<Option<T::Native>>,
{
    ...

    fn cumcount(&self, reverse: bool, count_nulls: bool) -> UInt32Chunked {
        let initial_value = 032;
        let mut chunked_array = match reverse {
            false => self
                .into_iter()
                .scan(initial_value, det_count)
                .collect_trusted(),
            true => self
                .into_iter()
                .rev()
                .scan(initial_value, det_count)
                .collect_reversed()
        };
        chunked_array.rename(self.name());
        chunked_array
    }
}
```

added [here](https://github.com/pola-rs/polars/blob/master/polars/polars-core/src/chunked_array/ops/cum_agg.rs#L73-L77). Is anyone able to help me understand what the problem is?